src/wp-includes/script-loader.php

```diff
@@ -360,20 +364,24 @@ function wp_default_packages_inline_scripts( $scripts ) {
 		),
 		'after'
 	);
+
+	$meta_key     = $wpdb->get_blog_prefix() . 'persisted_preferences';
+	$user_id      = get_current_user_ID();
+	$preload_data = get_user_meta( $user_id, $meta_key, true );
 	$scripts->add_inline_script(
-		'wp-data',
-		implode(
-			"\n",
-			array(
-				'( function() {',
-				'	var userId = ' . get_current_user_ID() . ';',
-				'	var storageKey = "WP_DATA_USER_" + userId;',
-				'	wp.data',
-				'		.use( wp.data.plugins.persistence, { storageKey: storageKey } );',
-				'	wp.data.plugins.persistence.__unstableMigrate( { storageKey: storageKey } );',
-				'} )();',
-			)
-		)
+		'wp-preferences',
+		sprintf(
+			'( function() {
+				var serverData = %s;
+				var userId = "%s";
+				var persistenceLayer = wp.preferencesPersistence.__unstableCreatePersistenceLayer( serverData, userId );
+				var preferencesStore = wp.preferences.store;
+				wp.data.dispatch( preferencesStore ).setPersistenceLayer( persistenceLayer );
+			} ) ();',
+			wp_json_encode( $preload_data ),
+			wp_json_encode( $user_id )
```

Comment:
I think it would be better to remove the call to `wp_json_encode` here and use `%d` instead of `%s` above. That we we avoid an unnecessary call to `wp_json_encode`, and we don't leave the reader second guessing whether `$user_id` is an integer or not (it is, always).

Reply:
`%d` makes sense.

For reference, I originally added `wp_json_encode` as per this comment - https://github.com/WordPress/gutenberg/pull/39795#discussion_r875143400.